dom/src/main/scala/org/scalajs/dom/ReadableStreamReader.scala

```diff
@@ -30,11 +30,8 @@ class ReadableStreamReader[+T](stream: ReadableStream[T]) extends js.Object {
     *
     * If the reader is active, the cancel method behaves the same as that for the associated stream. When done, it
     * automatically releases the lock.
-    *
-    * //todo determine type of reason
     */
-  // not actually sure what the return type is here
-  def cancel(reason: Any): js.Promise[Any] = js.native
+  def cancel[U](reason: js.UndefOr[U]): js.Promise[U] = js.native
```

Comment:
```suggestion
  def cancel[U](reason: js.UndefOr[U]): js.Promise[js.UndefOr[U]] = js.native
```

If the input is undefined, I assume the output can be as well?

Reply:
In that case:
```suggestion
  def cancel[U](reason: U): js.Promise[U] = js.native
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamDefaultReader/cancel does show a call with no arguments passed - and my guess that is the most common usage - so should it be `def cancel[U](reason: js.UndefOr[U] = js.undefined): js.Promise[js.UndefOr[U]] = js.native` then?

Reply:
Although `reason: U` will not return `js.undefined`, so my suggestion is also not ideal.

Reply:
What if we just provide two overloads of this method?

```scala
def cancel[U](reason: U): js.Promise[U] = js.native
def cancel(): js.Promise[Unit] = js.native
```

Reply:
Consider using 2 overloads, then:
```scala
  def cancel[U](reason: U): js.Promise[U] = js.native
  def cancel(): js.Promise[Unit] = js.native
```

Reply:
I was a bit late to also reply with the same message 😄